crates/seal-sdk/src/lib.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c), Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+pub mod types;
+
+pub use crypto::elgamal::{decrypt as elgamal_decrypt, genkey};
+pub use crypto::ibe::verify_user_secret_key as ibe_verify_user_secret_key;
+pub use crypto::ibe::PublicKey as IBEPublicKey;
+pub use crypto::{
+    seal_decrypt, seal_encrypt, EncryptedObject, EncryptionInput, IBEPublicKeys, IBEUserSecretKeys,
+};
+pub use types::{Certificate, ElGamalSecretKey, FetchKeyRequest, FetchKeyResponse};
+
+use crate::types::{ElGamalPublicKey, ElgamalVerificationKey};
+use chrono::{DateTime, Utc};
+use fastcrypto::ed25519::Ed25519PublicKey;
+use serde::{Deserialize, Serialize};
+use sui_sdk_types::ProgrammableTransaction;
+use tracing::debug;
+
+pub fn signed_message(
+    package_name: String,
+    vk: &Ed25519PublicKey,
+    creation_time: u64,
+    ttl_min: u16,
+) -> String {
+    let res = format!(
+        "Accessing keys of package {} for {} mins from {}, session key {}",
+        package_name,
+        ttl_min,
+        DateTime::<Utc>::from_timestamp((creation_time / 1000) as i64, 0)
+            .expect("tested that in the future"),
+        vk,
+    );
+    debug!("Signed message: {}", res.clone());
+    res
+}
+
+#[derive(Serialize, Deserialize)]
+struct RequestFormat {
+    ptb: Vec<u8>,
+    enc_key: Vec<u8>,
+    enc_verification_key: Vec<u8>,
+}
+
+pub fn signed_request(
```

Comment:
if this results in exactly the same vector, can't we use it from the key server already now?